obstacledetection/src/test/java/com/caneseeaproject/obstacledetection/ObstacleDetectorTest.kt

```diff
@@ -0,0 +1,37 @@
+package com.caneseeaproject.obstacledetection
+
+import androidx.lifecycle.Transformations.map
+import com.caneseeproject.obstacledetection.*
+import org.junit.Test
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.filterIsInstance
+
+class ObstacleDetectorTest {
+    private val testODTanslator = ODTranslator()
+    private var tokenizedMessage : ODReading? = null
+
+    lateinit var packagedMessage : String
+    var rangeControlTest = ODControl.RangeControl(50)
+
+    @Test
+    fun testTokenizingObstacleDistance() {
+        tokenizedMessage = testODTanslator.tokenize("${OBSTACLE_DISTANCE}_6.5")
+        println(tokenizedMessage)
+        assert(tokenizedMessage != null && tokenizedMessage is ODReading.ObstacleDistance)
+    }
```

Comment:
while you're at it, `assert` that the object actually contains `6.5`

Reply:
Tried for hours to, but i don't know how to deal with return type from `tokenize` as `ODReading.ObstacleDistance` not only `ODReading`.

Reply:
I will give it a try till I give up , until that time , don't give me a hand :D